### PR TITLE
[autogluon][build] AutoGluon 0.8.0 training containers

### PR DESCRIPTION
**Pull request by yinweisu**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [x] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,68 @@
+ARG PYTHON_VERSION=3.9.16
+
+FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/pytorch-training:1.13.1-cpu-py39-ubuntu20.04-sagemaker
+
+# PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK - see https://github.com/autogluon/autogluon/issues/2534
+# ENV PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK=1
```
Comment (gradientsky) [inline autogluon/training/docker/0.8/py3/Dockerfile.cpu:6]: should we completely remove this?

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu.os_scan_allowlist.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
+  "grpcio": [
```
Comment (gradientsky) [inline autogluon/training/docker/0.8/py3/Dockerfile.cpu.os_scan_allowlist.json:2]: `reason_to_ignore` is missing for each item. See example [here](https://github.com/aws/deep-learning-containers/blob/master/autogluon/training/docker/0.6/py3/Dockerfile.cpu.os_scan_allowlist.json#L30). Must be last entry in a block.

```
      "reason_to_ignore": "patch via AWS binaries in parent container"
```

File: autogluon/training/docker/0.8/py3/cu117/Dockerfile.gpu.os_scan_allowlist.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{
+  "grpcio": [
```
Comment (gradientsky) [inline autogluon/training/docker/0.8/py3/cu117/Dockerfile.gpu.os_scan_allowlist.json:2]: ditto `reason_to_ignore`

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu (hunk 1 above)
Comment (yinweisu) [reply autogluon/training/docker/0.8/py3/Dockerfile.cpu:6]: Yea, I just tried removing it and did a smoke test for timeseries, looks good

Comment (tonyhoo) [reply autogluon/training/docker/0.8/py3/Dockerfile.cpu:6]: We shall remove this as it is resolved as part of CUDA 1.13.1 upgrade

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+# Install dependencies
+RUN apt-get update \
+ && apt-get -y upgrade \
+ && apt-get install -y --no-install-recommends \
+ && apt-get autoremove -y \
+ # Install tesseract-ocr to support automm document classification
+ && apt-get install tesseract-ocr -y \
+ # Install rsync to support ray distributed training
+ && apt-get install rsync -y \
```
Comment (tonyhoo) [inline autogluon/training/docker/0.8/py3/Dockerfile.cpu:19]: Do we still need this for 0.8?

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+# update cryptography >=41.0.0: CVE-2023-2650. This comes from the base image
+# update tornado >= 6.3.2: CVE-2023-28370. This comes from the base image
+# update pyOpenSSL 23.2.0 so cryptography can be updated
+# pin mmengine to avoid MultiModalPredictor import crashing
+# pin yapf to resolve importlib-metadata conflict with sagemaker==2.150.0
+RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
+ && pip install --no-cache-dir -U setuptools wheel \
+ && pip install --no-cache-dir -U numpy numba \
+ && pip install --no-cache-dir -U autogluon==${AUTOGLUON_VERSION} \
+ && mim install -q mmcv \
```
Comment (tonyhoo) [inline autogluon/training/docker/0.8/py3/Dockerfile.cpu:36]: Shall we install 'pymupdf` as well as it is moved to extra dependency at [here](https://github.com/autogluon/autogluon/blob/master/multimodal/setup.py#L72C10-L72C17)

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu (hunk 4 above)
Comment (yinweisu) [reply autogluon/training/docker/0.8/py3/Dockerfile.cpu:19]: Yes, we will need it as long as we want to support distributed training with ray

File: autogluon/training/docker/0.8/py3/Dockerfile.cpu (hunk 5 above)
Comment (yinweisu) [reply autogluon/training/docker/0.8/py3/Dockerfile.cpu:36]: Good point. Let me add that

**Review by tonyhoo — DISMISSED** [review]

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**